### ISSUE TITLE
Web hook validation in 2.x+ not present anymore?

### ISSUE THREAD

**Issue by sandromastronardi** [cloudevents/sdk-csharp#243]
Related to fix #145 

I am trying to update my cloud Events sdk to the newest version with the fix included but I can't make the validation work anymore. Is the fix not even relevant anymore?

I used to have this:
  public async Task<HttpResponseMessage> HandleSubscriptionValidationEvent(HttpRequestMessage req)
          {
              if (!await req.IsWebHookValidationRequest())
              {
                  throw new NotSupportedException("The request is not a WebHook validation request!");
              }

...

            var response = await req.HandleAsWebHookValidationRequest((origin) =>
            {
                var setting = _settings.WebHookAllowedOrigin ?? Environment.GetEnvironmentVariable("WebHook-Request-Origin");
                if (string.IsNullOrEmpty(setting) || setting == "*")
                {
                    return true;
                }
                else
                {
                    return setting == origin;
                }
            }, (o) =>
            {
                return _settings.WebHookAllowedRate ?? Environment.GetEnvironmentVariable("WebHook-Request-Rate") ?? "*";
            });
            return response;

But the IsWebHookValidationRequest and HandleAsWebHookValidationRequest are gone, how do i do this now?

Sandro

**Comment by jskeet** [comment]
Okay, I've found it - we did indeed remove all webhook code in https://github.com/cloudevents/sdk-csharp/commit/a91d138b96a4e31003e939409f906ae25b2f68d6, to avoid a later breaking change being needed due to https://github.com/cloudevents/spec/issues/781. This has now been fixed by https://github.com/cloudevents/spec/pull/865, which changes the header from Origin to WebHook-Request-Origin, which appears to be the header the code was already using. So I *may* be able to just revert that earlier commit and do another release, but I'll need to check it through very carefully - I doubt that it'll be within the next couple of weeks, I'm afraid.

Until then, I *suspect* you could just copy the deleted code into your own project (with appropriate attribute/licences) and use it - I doubt that it uses any internals from CloudNative.CloudEvents (although I haven't checked in detail).

**Comment by jskeet** [comment]
Yes, the library 2.0 is still CloudEvents 1.0 - there isn't a newer major version of the CloudEvents spec. It's only a new major version because the .NET API took breaking changes (and it's now signed). If Azure only produces genuinely valid CloudEvents (and can consume genuinely valid ones) then you should be fine.

I would suggest updating to the latest library, then putting the fixed code in a separate source file in your own project, using the existing namespaces, so you can just adopt 2.7.0 (or whatever it ends up being) by updating the reference and deleting the file.

**Comment by sandromastronardi** [comment]
Ok, great. It adhered the standards mostly (except for some missing headers that I add along they way when they are missing to be able to use the validation).
So I'll move it locally for now.
